### ISSUE TITLE
GridView not working anymore since upgrade to RN 0.58.4

### ISSUE THREAD

**Issue by jonathangreco** [saleel/react-native-super-grid#81]
I've an issue with this code :

```

              <FlatGrid
                ref={(grid) => {
                  this.gridView = grid
                }}
                itemDimension={Util.isPortrait() ? Number.parseInt(Dimensions.get('window').width * 0.96) : Number.parseInt(Dimensions.get('window').width * 0.40)}
                items={templates}
                style={styles.gridView}
                spacing={5}
                renderItem={this._renderRow}
              />

```

When itemDimension is a float and vary with onLayout events, your code throw an error.

The error is located here : https://github.com/saleel/react-native-super-grid/blob/master/utils.js#L27

itemsPerRow can be 0 in my case when switching from landscape to portrait. especially with the snippet given. Division by zero is prohibited as we all know. 

I replaced your code by
`const containerDimension = availableDimension / (itemsPerRow > 0) ? itemsPerRow : 1;`

But it's far from perfect, I really need your insight on this.

**Comment by jonathangreco** [comment]
The problem is inf the file 

`FlatGrid.js`

```

    return (
      <View style={[rowStyle, additionalRowStyle]}>
        {rowItems.map((item, i) => (
          <View
            key={`item_${(rowIndex * itemsPerRow) + i}`}
            style={[containerStyle, itemContainerStyle]}
          >
            {renderItem({
              item,
              index: (rowIndex * itemsPerRow) + i,
              separators,
              rowIndex,
            })}
          </View>
        ))}
      </View>
    );
```
If I remove styles attribute the red screen go away.

**Comment by jonathangreco** [comment]
It seems 

```
Util.isPortrait() ? Dimensions.get('window').width * 0.96 : Dimensions.get('window').width * 0.40
```

Give infinity since 0.58.4 why ?

**Comment by jonathangreco** [comment]
I manage to fix this.

But my issue become more tricky now :


On your containerStyle 

```
  let containerStyle = {
    flexDirection: 'column',
    justifyContent: 'center',
    width: fixed ? itemDimension : (containerDimension - spacing),
    marginRight: fixed ? fixedSpacing : spacing,
  };
```
I got the redscreen of death
```
ExceptionsManager.js:82 Malformed calls from JS: field sizes are different.

[[26,43,43,43,43,43,56,32,32,32,32,32,32,26,32,32,32,32,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,43,12,12,12,53,44,43,12,69,12,6],[0,5,5,3,5,5,2,1,1,1,1,1,1,1,1,1,1,1,10,2,2,2,11,11,11,10,2,2,2,2,11,11,11,11,11,4,0,0,0,4,1,4,0,0,1,1],[[248,50,1550243837737,false],[469,"RCTActivityIndicatorView",41,{"size":"small","animating":true,"hidesWhenStopped":true,"height":20,"width":20,"color":4288256409}],[473,"RCTView",41,{"alignItems":"center","justifyContent":"center"}],[473,[469]],[477,"RCTView",41,{"height":0}],[1],[1],[1],[1],[1],[1],[1],[248],[1],[1],[1],[1],[14,185],[13,14],[12,13],[4,12],[12],[13],[14],[18,283],[17,18],[16,17],[15,16],[4,15],[4],[15],[16],[17],[18],[57,[],[],[],[],[0]],["Invariant Violation: [475,\"RCTView\",41,{\"flexDirection\":\"column\",\"justifyContent\":\"center\",\"width\":\"<<Infinity>>\",\"marginRight\":5}] is not usable as a native method argument\n\nThis error is located at:\n    in RCTView (at View.js:45)\n    in View (at FlatGrid.js:77)\n 
```
This is happenning only when On Layout from landscape to portait...

**Comment by jonathangreco** [comment]
@saleel Hi can you take a look at this problem please ?

**Comment by saleel** [comment]
Yes..checking now

**Comment by jonathangreco** [comment]
@saleel 
I had this error in a new version of RN yes as my title suggest it. They made some changes.

The error I have is a red screen of death with 

`ExceptionsManager.js:82 Malformed calls from JS: field sizes are different.`

**Comment by jonathangreco** [comment]
@saleel Hi do you need more informations ?

**Comment by saleel** [comment]
You are right. I was able to reproduce this with new version of RN. I have fixed this and released a new version 3.0.2. Please install it from npm. You can see the code change I made [here](https://github.com/saleel/react-native-super-grid/commit/1a5588f2468d4bc5cad682c4a726e170f46d1196).

**Comment by jonathangreco** [comment]
@saleel  it works better :)
